src/screens/HyperSwitch/SwitchMerchant/SwitchMerchantListHook.res

```diff
@@ -0,0 +1,19 @@
+let useFetchSwitchMerchantList = () => {
+  open APIUtils
+  let fetchDetails = useGetMethod()
+  let setSwitchMerchantListAtom = HyperswitchAtom.switchMerchantListAtom->Recoil.useSetRecoilState
+  let url = getURL(~entityName=USERS, ~userType=#SWITCH_MERCHANT, ~methodType=Get, ())
+
+  async _ => {
+    try {
+      let res = await fetchDetails(url)
+      let typedValueOfResponse = res->SwitchMerchantUtils.convertListResponseToTypedResponse
+      setSwitchMerchantListAtom(._ => typedValueOfResponse)
+    } catch {
+    | Exn.Error(e) => {
+        let err = Exn.message(e)->Option.getOr("Failed to Fetch!")
```

Comment:
Should the exception be more specific?
Ignore this suggestion if you'd like to push logs here instead (in a later PR)

Reply:
We would want to have analytics over exceptions to identify and mitigate issues